### PR TITLE
feat: add slack oidc provider

### PR DESCRIPTION
**Pull request by J0**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

Adds Slack provider.

BE PR: https://github.com/supabase/infrastructure/pull/18656

Auth Changes: https://github.com/supabase/auth/pull/1591/files

### CODE REVIEW

File: apps/studio/components/interfaces/Auth/AuthProvidersFormValidation.tsx [hunk 1]
```diff
@@ -1053,6 +1053,44 @@ const EXTERNAL_PROVIDER_LINKEDIN_OIDC = {
...
+    EXTERNAL_SLACK_OIDC_CLIENT_ID: {
+      title: 'API Key',
+      type: 'string',
+    },
+    EXTERNAL_SLACK_OIDC_SECRET: {
+      title: 'API Secret Key',
+      type: 'string',
+      isSecret: true,
+    },
+  },
```
Comment (hf) [inline apps/studio/components/interfaces/Auth/AuthProvidersFormValidation.tsx:1074]: Since when do we do this?

Comment (J0) [reply apps/studio/components/interfaces/Auth/AuthProvidersFormValidation.tsx:1074]: Do you mean the validation schema? Not too sure - referencing what we did for [LinkedIn OIDC](https://github.com/supabase/supabase/pull/18225/files#diff-c842696488522c2e7915e506fc71296714db67c2f9c483a3dcf903d88c267a27)

There's more bits to update, including docs but wanted to get a minimal version out for testing.

**Comment by kangmingtay** [comment]
@J0 added the deprecation notice and config updates in [this PR](https://github.com/supabase/supabase/pull/27864), i think we can close this one - my bad, i only realised there was an open PR for this after I opened the PR 🤦

**Comment by J0** [comment]
Sounds good, no worries

**Pull request closed without merging**